python/instrumentation/openinference-instrumentation-boto/src/openinference/instrumentation/boto/__init__.py

```diff
@@ -0,0 +1,169 @@
+
+
+class BufferedStreamingBody(StreamingBody):  # type: ignore
+    _raw_stream: IO[bytes]
+
+    def __init__(self, raw_stream: IO[bytes], content_length: int) -> None:
+        super().__init__(raw_stream, content_length)
+        self._buffer: Optional[io.IOBase] = None
+
+    def read(self, amt: Optional[int] = None) -> bytes:
+        if self._buffer is None:
+            self._buffer = io.BytesIO(self._raw_stream.read())
+
+        output: bytes = self._buffer.read(amt)
+        return output
+
+    def reset(self) -> None:
+        # Reset the buffer to enable reading the stream again
+        if self._buffer is not None:
+            self._buffer.seek(0)
+
+
+def _client_creation_wrapper(tracer: Tracer) -> Callable[[CallableType], CallableType]:
+    def _client_wrapper(
+        wrapped: CallableType,
+        instance: Optional[Any],
+        args: Tuple[Any, ...],
+        kwargs: Dict[str, Any],
+    ) -> BaseClient:
+        @wraps(wrapped)
+        def create_instrumented_client(*args: Any, **kwargs: Any) -> BaseClient:
+            """Instruments and calls every function defined in TO_WRAP."""
+            client: BaseClient = wrapped(*args, **kwargs)
+            if context_api.get_value(_SUPPRESS_INSTRUMENTATION_KEY):
+                return client
+
+            call_signature = signature(wrapped)
+            bound_arguments = call_signature.bind(*args, **kwargs)
+            bound_arguments.apply_defaults()
+
+            if bound_arguments.arguments.get("service_name") == "bedrock-runtime":
+                client = cast(InstrumentedClient, client)
+                client._unwrapped_invoke_model = client.invoke_model
+                client.invoke_model = _model_invocation_wrapper(tracer)(client)
+                return client
+            return client
+
+        return create_instrumented_client(*args, **kwargs)
+
+    return _client_wrapper  # type: ignore
+
+
+def _model_invocation_wrapper(tracer: Tracer) -> Callable[[InstrumentedClient], Callable[..., Any]]:
+    def _invocation_wrapper(wrapped_client: InstrumentedClient) -> Callable[..., Any]:
+        """Instruments a bedrock client's `invoke_model` method."""
+
+        @wraps(wrapped_client.invoke_model)
+        def instrumented_response(*args: Any, **kwargs: Any) -> Dict[str, Any]:
+            with tracer.start_as_current_span("bedrock.invoke_model") as span:
+                response = wrapped_client._unwrapped_invoke_model(*args, **kwargs)
+                response["body"] = BufferedStreamingBody(
+                    response["body"]._raw_stream, response["body"]._content_length
+                )
+                if raw_request_body := kwargs.get("body"):
+                    request_body = json.loads(raw_request_body)
+                response_body = json.loads(response.get("body").read())
+                response["body"].reset()
+
+                prompt = request_body.pop("prompt")
+                invocation_parameters = request_body
```

Comment:
The value for the `invocation_parameters` attribute is JSON string.
```suggestion
                invocation_parameters = json.dumps(request_body)
```